.github/workflows/build.yml

```diff
@@ -7,44 +7,138 @@ on:
   workflow_dispatch:
 
 jobs:
-  build:
+  build-linux:
     timeout-minutes: 30
+    runs-on: ubuntu-latest
     strategy:
       fail-fast: false
       matrix:
-        os: ["ubuntu-latest"]
-        generator: ["Unix Makefiles", "Ninja"]
+        setup:
+          - { c: "gcc",   cxx: "g++",     fortran: "gfortran", generator: "Unix Makefiles" }
+          - { c: "clang", cxx: "clang++", fortran: "gfortran", generator: "Ninja"}
 
-        include:
-          - os: ubuntu-latest
-            exe-ext: ""
-            metis-lib: "-DMOO_METIS_LIB=/usr/lib/x86_64-linux-gnu/libmetis.so.5.1.0"
-            lapack-lib: ""
+    steps:
+      - uses: actions/checkout@v5
+        with:
+          submodules: recursive
+
+      - name: Install Dependencies
+        run: |
+          sudo apt-get update -qq
+          sudo apt-get -qqy install libmetis-dev liblapack-dev gfortran
+
+      - name: Configure
+        run: |
+          cmake -S . -B build \
+            -DCMAKE_INSTALL_PREFIX=install \
+            -DCMAKE_C_COMPILER=${{ matrix.setup.c }} \
+            -DCMAKE_CXX_COMPILER=${{ matrix.setup.cxx }} \
+            -DCMAKE_Fortran_COMPILER=${{ matrix.setup.fortran }} \
+            -DMOO_WITH_GDOPT_EXAMPLE=ON \
+            -G "${{ matrix.setup.generator }}"
+
+      - name: Build
+        run: cmake --build build --target install -- -j$(nproc)
+
+      - name: Run test
+        run: ./build/gdopt
+
+      - uses: actions/upload-artifact@v4
+        with:
+          name: install-linux-${{ matrix.setup.c }}-${{ matrix.setup.generator }}
+          path: install/
+
+  build-windows:
+    timeout-minutes: 30
+    runs-on: windows-latest
+    strategy:
+      fail-fast: false
```

Comment:
See my comment in https://github.com/AMIT-HSBI/MOO/issues/10#issuecomment-3333418791

I would enable `fail-fast` for Windows and macOS. 

```diff
- fail-fast: false
```